src/FormTaxonomies/resources/form-builder/form-categories.tsx

```diff
@@ -0,0 +1,54 @@
+import {getAvailableFormCategories, getInitialFormCategories} from "./windowData";
+import {buildTermsTree} from "./utils/terms";
+import {CheckboxControl} from "@wordpress/components";
+import {decodeEntities} from "@wordpress/html-entities";
+
+
+const FormCategorySetting = ({settings, setSettings}) => {
+    const {
+        formCategories = getInitialFormCategories(),
+    } = settings;
+
+    const categoryTree = buildTermsTree(getAvailableFormCategories());
```

Comment:
Looks like you can move this outside of your component to avoid side effects of re-renders as it is working with data from the window that won't change.

Reply:
The data doesn't seem to be available when called outside of the component, so I memoized it, see [0f00bb7](https://github.com/impress-org/givewp/pull/7463/commits/0f00bb708095850f4be64e4102ba2d0d1a728d32).

Reply:
Okay that works too 👍